cmd/raidz_test/raidz_test.c

```diff
@@ -0,0 +1,800 @@
+			eret++;
+			if (opts->rto_v > D_INFO) {
+				PRINT("data block [%d] different!\n", i);
+			}
+		}
+	}
+	return (eret);
+}
+
+static void
+corrupt_colums(raidz_map_t *rm, const int *tgts, const int cnt)
+{
+	int i;
+	int *dst;
+	raidz_col_t *col;
+
+	for (i = 0; i < cnt; i++) {
+		col = &rm->rm_col[tgts[i]];
+		dst = col->rc_data;
+		for (i = 0; i < col->rc_size / sizeof (int); i++)
+			dst[i] = rand();
+	}
+}
+
+void
+init_zio_data(zio_t *zio)
+{
+	int i;
+	int *dst = (int *) zio->io_data;
+
+	for (i = 0; i < zio->io_size / sizeof (int); i++) {
+		dst[i] = rand_data[i];
+	}
+}
+
+static void
+fini_raidz_map(zio_t **zio, raidz_map_t **rm)
+{
+	vdev_raidz_map_free(*rm);
+	raidz_free((*zio)->io_data, (*zio)->io_size);
+	umem_free(*zio, sizeof (zio_t));
+
+	*zio = NULL;
+	*rm = NULL;
+}
+
+static int
+init_raidz_golden_map(raidz_test_opts_t *opts, const int parity)
+{
+	int err = 0;
+	zio_t *zio_test;
+	raidz_map_t *rm_test;
+	const size_t total_ncols = opts->rto_dcols + parity;
+
+	if (opts->rm_golden) {
+		fini_raidz_map(&opts->zio_golden, &opts->rm_golden);
+	}
+
+	opts->zio_golden = umem_zalloc(sizeof (zio_t), UMEM_NOFAIL);
+	zio_test = umem_zalloc(sizeof (zio_t), UMEM_NOFAIL);
+
+	opts->zio_golden->io_offset = zio_test->io_offset = opts->rto_offset;
+	opts->zio_golden->io_size = zio_test->io_size = opts->rto_dsize;
+
+	opts->zio_golden->io_data = raidz_alloc(opts->rto_dsize);
+	zio_test->io_data = raidz_alloc(opts->rto_dsize);
+
+	init_zio_data(opts->zio_golden);
+	init_zio_data(zio_test);
+
+	VERIFY0(vdev_raidz_impl_set("original"));
+
+	opts->rm_golden = vdev_raidz_map_alloc(opts->zio_golden,
+	    opts->rto_ashift, total_ncols, parity);
+	rm_test = vdev_raidz_map_alloc(zio_test,
+	    opts->rto_ashift, total_ncols, parity);
+
+	VERIFY(opts->zio_golden);
+	VERIFY(opts->rm_golden);
+
+	vdev_raidz_generate_parity(opts->rm_golden);
+	vdev_raidz_generate_parity(rm_test);
+
+	/* sanity check */
+	err |= cmp_data(opts, rm_test);
+	err |= cmp_code(opts, rm_test, parity);
+
+	if (err)
+		PRINT(DBLSEP "initializing the golden copy ... [FAIL]!\n");
+
+	/* tear down raidz_map of test zio */
+	fini_raidz_map(&zio_test, &rm_test);
+
+	return (err);
+}
+
+static raidz_map_t *
+init_raidz_map(raidz_test_opts_t *opts, zio_t **zio, const int parity)
+{
+	unsigned err = 0;
+	raidz_map_t *rm = NULL;
+	const size_t alloc_dsize = opts->rto_dsize;
+	const size_t total_ncols = opts->rto_dcols + parity;
+	const int ccols[] = { 0, 1, 2 };
+
+	VERIFY(zio);
+	VERIFY(parity <= 3 && parity >= 1);
+
+	*zio = umem_alloc(sizeof (zio_t), UMEM_NOFAIL);
+
+	bzero(*zio, sizeof (zio_t));
+
+	(*zio)->io_offset = 0;
+	(*zio)->io_size = alloc_dsize;
+	(*zio)->io_data = raidz_alloc(alloc_dsize);
+	init_zio_data(*zio);
+
+	rm = vdev_raidz_map_alloc(*zio, opts->rto_ashift,
+		total_ncols, parity);
+	VERIFY(rm);
+
+	/* Make sure code columns are destroyed */
+	corrupt_colums(rm, ccols, parity);
+
+
+	if (err) {
+		PRINT("ERROR!\n");
+		exit(-1);
+	}
```

Comment:
How can this ever happen?  This conditional and `err` variable look like they should be removed entirely.